File: app/src/main/java/com/stedi/randomimagegenerator/app/presenter/interfaces/GenerationStepsPresenter.java
package com.stedi.randomimagegenerator.app.presenter.interfaces;

import com.stedi.randomimagegenerator.app.presenter.interfaces.core.Presenter;
import com.stedi.randomimagegenerator.app.presenter.interfaces.core.UI;

public interface GenerationStepsPresenter extends Presenter<GenerationStepsPresenter.UIImpl> {
    void setIsNew(boolean isNew);

    void release();

    interface UIImpl extends UI {
        void showFirstStep();

        void showFinishStep();
    }
}

File: app/src/main/java/com/stedi/randomimagegenerator/app/presenter/interfaces/GenerationStepsPresenter.kt
package com.stedi.randomimagegenerator.app.presenter.interfaces

import com.stedi.randomimagegenerator.app.presenter.interfaces.core.Presenter
import com.stedi.randomimagegenerator.app.presenter.interfaces.core.UI

interface GenerationStepsPresenter : Presenter<GenerationStepsPresenter.UIImpl> {
    fun setIsNew(isNew: Boolean)

    fun release()

    interface UIImpl : UI {
        fun showFirstStep()

        fun showFinishStep()
    }
}
